metricflow/test/snapshots/test_dataflow_to_sql_plan.py/SqlQueryPlan/DuckDbSqlClient/test_derived_metric__plan0_optimized.sql

```diff
@@ -0,0 +1,57 @@
+-- Compute Metrics via Expressions
+SELECT
+  metric_time
+  , (bookings - ref_bookings) / bookings AS non_referred_bookings_pct
+FROM (
+  -- Combine Metrics
+  SELECT
+    COALESCE(subq_19.metric_time, subq_20.metric_time) AS metric_time
+    , subq_19.ref_bookings AS ref_bookings
+    , subq_20.bookings AS bookings
+  FROM (
+    -- Aggregate Measures
+    -- Compute Metrics via Expressions
+    SELECT
+      metric_time
+      , SUM(referred_bookings) AS ref_bookings
+    FROM (
+      -- Read Elements From Data Source 'bookings_source'
+      -- Metric Time Dimension 'ds'
+      -- Pass Only Elements:
+      --   ['referred_bookings', 'metric_time']
+      SELECT
+        ds AS metric_time
+        , CASE WHEN referrer_id IS NOT NULL THEN 1 ELSE 0 END AS referred_bookings
+      FROM (
+        -- User Defined SQL Query
+        SELECT * FROM ***************************.fct_bookings
+      ) bookings_source_src_10001
+    ) subq_13
+    GROUP BY
+      metric_time
+  ) subq_19
+  FULL OUTER JOIN (
```

Comment:
Oh, huh.... should this be a FULL OUTER join?

With multi-measure ratio and expr metrics we do an inner join, because those don't make sense when NULL inputs get involved - you end up with a bunch of uncontrolled failure states. For example:

https://github.com/transform-data/metricflow/blob/main/metricflow/test/snapshots/test_dataflow_to_sql_plan.py/SqlQueryPlan/DuckDbSqlClient/test_compute_metrics_node_ratio_from_multiple_data_sources__plan0_optimized.sql#L40